### PR TITLE
[tools/vanity]: tools/vanity is missing .coveragerc file

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: code coverage numbers aren't being calculated correctly due to missing .coveragerc file
solution: add .coveragerc file in line with other python projects
          add tests for __main__

### CODE REVIEW

**Review by Wayonb — APPROVED** [review]
Looks good. 👍🏾 

If you want to exclude the one line see https://coverage.readthedocs.io/en/latest/excluding.html

File: tools/vanity/vanity/__main__.py [hunk 1]
```diff
@@ -53,8 +53,9 @@ def main():
...
 	finally:
 		for outfile in printer.outfiles:
-			outfile.close()
+			if '<stdout>' != outfile.name:
+				outfile.close()
 
 
 if '__main__' == __name__:
-	main()
+	main(sys.argv[1:])
```
Comment (gimre-xymcity) [inline tools/vanity/vanity/__main__.py:61]: hmm, why is this needed?

**Review by gimre-xymcity — DISMISSED** [review]

File: tools/vanity/vanity/__main__.py (hunk 1 above)
Comment (Jaguar0625) [reply tools/vanity/vanity/__main__.py:61]: because main() now takes a parameter so it can be called by tests (so argparse doesn't operate on global)
